x/superfluid/keeper/stake.go

```diff
@@ -231,53 +281,31 @@ func (k Keeper) SuperfluidDelegate(ctx sdk.Context, sender string, lockID uint64
 	if err != nil {
 		return err
 	}
-	k.ak.SetAccount(ctx, authtypes.NewBaseAccount(mAddr, nil, 0, 0))
+	// TODO: @Dev added this hasAccount gating, think through if theres an edge case that makes it not right
+	if !k.ak.HasAccount(ctx, mAddr) {
+		// TODO: Why is this a base account, not a module account?
```

Comment:
Aren't intermediary accounts supposed to be base account not module account?

Reply:
oh is there a reason for base vs module?

Reply:
Yeah BaseAccounts are a standard account.

The questions I think of for ModuleAccounts vs BaseAccount is:
* (Primary Q) Is there a private key that can control it
* should it have different permissions than a normal account (mint/burn of an asset)
* Should people just send coins to it

Reply:
It is ok to have many module accounts, we just have to do some API hacks IIRC, since the base SDK doesn't give us the best methods to do it